From: Jiebing Li <jiebing.li@intel.com>
Subject: usb: gadget: ffs: avoid disable ep twice

When FFS gadget function is disabled, gadget driver tries to
disable the endpoints twice, which triggers warning message
in device driver.

This patch ensures that usb_ep_disable() is not called again
after the endpoint has been disabled.

Signed-off-by: Jiebing Li <jiebing.li@intel.com>
Signed-off-by: David Cohen <david.a.cohen@intel.com>

---
 drivers/usb/gadget/f_fs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux.orig/drivers/usb/gadget/f_fs.c
+++ linux/drivers/usb/gadget/f_fs.c
@@ -1312,7 +1312,7 @@ static void ffs_func_eps_disable(struct
 	spin_lock_irqsave(&func->ffs->eps_lock, flags);
 	do {
 		/* pending requests get nuked */
-		if (likely(ep->ep))
+		if (ep->ep && epfile->ep)
 			usb_ep_disable(ep->ep);
 		epfile->ep = NULL;
 
